contracts/AelinUpFrontDeal.sol

```diff
@@ -218,7 +223,7 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
         totalPurchasingAccepted += purchaseTokenAmount;
         purchaseTokensPerUser[msg.sender] += purchaseTokenAmount;
 
-        uint8 underlyingTokenDecimals = IERC20Decimals(dealData.underlyingDealToken).decimals();
+        uint8 underlyingTokenDecimals = IERC20Extended(dealData.underlyingDealToken).decimals();
         uint256 poolSharesAmount;
```

Comment:
we can improve this a bit by removing uint256 poolSharesAmount with the line below adding the logic; and just doing uint256 poolSharesAmount = (purchaseTokenAmount * 10**underlyingTokenDecimals) / _purchaseTokenPerDealToken;